api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -141,44 +145,72 @@ public Cohort(String commaSeparatedIds) {
 	 */
 	public String getCommaSeparatedPatientIds() {
 		StringBuilder sb = new StringBuilder();
-		for (Iterator<Integer> i = getMemberIds().iterator(); i.hasNext();) {
-			sb.append(i.next());
-			if (i.hasNext()) {
-				sb.append(",");
-			}
+        for (CohortMembership member : getMembers()) {
+			sb.append(member.getPatient().getPatientId());
+			sb.append(",");
 		}
 		return sb.toString();
 	}
-	
+
+	// TODO: Compare patient objects instead of IDs after removal of deprecated methods
 	public boolean contains(Patient patient) {
-		return getMemberIds() != null && getMemberIds().contains(patient.getPatientId());
+		return getMembers() != null && getMembers().stream()
+				.anyMatch(m -> m.getPatient().getPatientId().equals(patient.getPatientId()));
 	}
 	
 	public boolean contains(Integer patientId) {
-		return getMemberIds() != null && getMemberIds().contains(patientId);
+		return getMembers() != null && getMembers().stream()
+				.anyMatch(m -> m.getPatient().getPatientId().equals(patientId));
 	}
-	
+
+	public boolean contains(CohortMembership cohortMembership) {
```

Comment:
I would think the service methods should just call add or remove on the Cohort object and under the hood these Cohort methods silently ignore duplicates, do you intend to do anything in the service in case this fails?